Source/resources/mongoDB/IMongoDBResource.ts

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+import { GetMongoDbResponse, GetRequest } from '@dolittle/runtime.contracts/Resources/Resources_pb';
+import { Cancellation } from '@dolittle/sdk.resilience';
+import { Resource, ResourceName } from '../index';
+import { MongoDBConnectionString } from './MongoDBConnectionString';
+
+/**
+ * Represents an implementation of {@link Resource} for MongoDB.
+ */
+export abstract class IMongoDBResource extends Resource<GetRequest, GetMongoDbResponse> {
+    /**
+     * Gets the the MongoDB connection string.
+     * @param {Cancellation} [cancellation] The optional {@link Cancellation}.
+     */
+    abstract getConnectionString(cancellation?: Cancellation): Promise<MongoDBConnectionString>;
```

Comment:
Is it going to be annoying for people to use the ConceptAs type from this API?

Reply:
I'm not sure. I think that I would have liked it if I were the consumer, but I'm also biased 😋